### PR TITLE
rafctor LookupCluster

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

1. remove duplicate code
2. introduce new metric for lookup fail

### CODE REVIEW

File: pilot/pkg/model/monitoring.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import "istio.io/pkg/monitoring"
+
+func init() {
+	monitoring.MustRegister(
+		providerLookupClusterFailures,
+	)
+}
+
+var providerLookupClusterFailures = monitoring.NewSum(
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/monitoring.go:25]: I think Gauge is better, we can record which cluster look up failed rather than always increasing the num

Comment (zirain) [reply pilot/pkg/model/monitoring.go:25]: when should pilot decrease the count?

Comment (hzxuzhonghu) [reply pilot/pkg/model/monitoring.go:25]: Please refer to envoyfilter patching metric, the sum number here is meaningless, it oculd be unlimited big

Comment (zirain) [reply pilot/pkg/model/monitoring.go:25]: For a counter, user should care about `rate` or `increase`.
if you are talking about `pilot_envoy_filter_status`, it's disabled by default.

Comment (zirain) [reply pilot/pkg/model/monitoring.go:25]: cc @ramaraochavali , do you know why it's disabled?

Comment (ramaraochavali) [reply pilot/pkg/model/monitoring.go:25]: It is disabled because of possible performance impact. People who want will only enable it in dev envs

Comment (zirain) [reply pilot/pkg/model/monitoring.go:25]: After take a look at `istio.io/pkg/monitoring`, it's impossible to reset all metrics to `0` when services or telemetries chaged.
cc @hzxuzhonghu

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**